From e1f8d3bc642cf1268bad579d267aeb6702f24834 Mon Sep 17 00:00:00 2001
From: Luis Henriques <luis@igalia.com>
Date: Wed, 18 Jun 2025 13:41:52 +0100
Subject: [PATCH] Workaround the lack of support for rename2() in musl

This should be fixed in the next musl release because that support has
been already added with commit:

05ce67fea99c ("add renameat2 linux syscall wrapper")
---
 example/memfs_ll.cc | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/example/memfs_ll.cc b/example/memfs_ll.cc
index 603885017d4c..db24dc5de4e9 100644
--- a/example/memfs_ll.cc
+++ b/example/memfs_ll.cc
@@ -34,6 +34,15 @@
 
 #define DIV_ROUND_UP(n, d) (((n) + (d) - 1) / (d))
 
+/*
+ * Workaround the lack of support for renameat2() for current musl version.
+ */
+#if defined(_GNU_SOURCE)
+#define RENAME_NOREPLACE (1 << 0)
+#define RENAME_EXCHANGE  (1 << 1)
+#define RENAME_WHITEOUT  (1 << 2)
+#endif
+
 class Inodes;
 class Inode;
 class Dentry;
